### ISSUE TITLE
Binary classification

### ISSUE THREAD

**Issue by iammiori** [iammiori/AI_19_SS_CLASS#4]
- Binary Label Encoding -> 0 / 1
- linear regression 쓰면, hypothesis 가 간단하지만, hypothesis는 1보다 크고, 보다 작을 수 있어.
- 하지만 우리는 0, 1 사이의 데이터 원해 => **logistic**

**Comment by iammiori** [comment]
### sigmoid

**sigmoid**
> : curved in two directions (like S) 
> : logistic function := sigmoid function
> sigmoid 때문에 h(x) 는 [0,1] 로 bound 됨

**Comment by iammiori** [comment]
### hypothesis

> = 1/ (1+ e^(-W^T  * X))

**Comment by iammiori** [comment]
### cost function

> - 기존 linear regression 에 cost function 적용하면 local minima 에 빠짐
> - cost function : 우리의 예측값이 얼마나 정답에 가까운가를 측정하는 척도
> - 말인즉슨, 정답에 가까워 질수록 cost function 값 작게
> - 정답에서 멀어질 수록 cost function 값 크게
> - Y=1 일때는 -log(H(x))
> - Y=0 일때는 -log(1-H(x))
> - cost = -1/m 시그마 {ylog(H(x)) + (1-y)log(1-H(x))}

**Comment by iammiori** [comment]
### Gradient Decent Alogrithm

> code 로 나타내면

`cost = tf.reduce_mean(-tf.reduce_mean(Y*tf.log(hypothesis) + (1-Y)*tf.log(1-yhopthesis))`

> minimize 시키면
```
a = tf.Variable(0.1)
optimizer = tf.train.GradientDescentOptimizer(a)
train = optimizer.minimize(cost)
```

**Comment by iammiori** [comment]
matrix 에서도 사용이 가능

`H(X) = XW`

**Comment by iammiori** [comment]
### Softmax

> 모든 값이 [0,1]
> 전체합이 1 (확률정규화)

**Comment by iammiori** [comment]
### One-hot encoding

> argmax 써서 확률을 1, 0으로 바꿔줘

### Cross-entropy

> D(S,L) = - 시그마 Li log(Si)
